feature-libs/asm/components/customer-list/customer-list.component.ts

```diff
@@ -0,0 +1,243 @@
+import { Component, OnInit } from '@angular/core';
+import { AsmConfig, AsmService } from '@spartacus/asm/core';
+import {
+  CustomerListColumnActionType,
+  CustomerListsPage,
+  CustomerSearchOptions,
+  CustomerSearchPage,
+} from '@spartacus/asm/root';
+import { SortModel, TranslationService, User } from '@spartacus/core';
+import {
+  BREAKPOINT,
+  BreakpointService,
+  FocusConfig,
+  ICON_TYPE,
+  ModalService,
+} from '@spartacus/storefront';
+import { combineLatest, Observable, of } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { CustomerListActionEvent } from './customer-list.model';
+
+@Component({
+  selector: 'cx-customer-list',
+  templateUrl: './customer-list.component.html',
+})
+export class CustomerListComponent implements OnInit {
+  private PAGE_SIZE = 5;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'customer-list-selector',
+    focusOnEscape: true,
+  };
+
+  iconTypes = ICON_TYPE;
+
+  BREAKPOINT = BREAKPOINT;
+
+  selectedUserGroupId: string | undefined;
+
+  customerSearchPage$: Observable<CustomerSearchPage | null>;
+
+  customerListsPage$: Observable<CustomerListsPage | undefined>;
+
+  selectedCustomer: User;
+
+  currentPage = 0;
+
+  maxPage = 0;
+
+  loaded = false;
+
+  sorts: SortModel[] | null;
+
+  sortCode: string | undefined;
+
+  breakpoint$: Observable<BREAKPOINT>;
+
+  customerListConfig = this.asmConfig?.asm?.customerList;
+
+  constructor(
+    protected modalService: ModalService,
+    protected asmService: AsmService,
+    protected breakpointService: BreakpointService,
+    protected asmConfig: AsmConfig,
+    protected translation: TranslationService
+  ) {
+    this.breakpoint$ = this.getBreakpoint();
+  }
+
+  ngOnInit(): void {
+    this.customerListsPage$ = this.asmService
+      .getCustomerLists()
+      .pipe(
+        filter((queryState) => queryState.loading === false),
+        map((queryState) => {
+          return queryState.data;
+        })
+      )
+      .pipe(
+        map((customerListsPage) => customerListsPage),
+        filter(() => true),
+        // set the first value of this.customerListsPage$ to be selected
+        tap((result) => {
+          if (!this.selectedUserGroupId) {
+            this.selectedUserGroupId = result?.userGroups?.[0].uid;
+            this.sorts = null;
+            this.fetchCustomers();
+          }
+        })
+      );
+  }
+
+  fetchCustomers(): void {
+    const pageSize =
+      this.asmConfig.asm?.customerList?.pageSize ?? this.PAGE_SIZE;
+    if (this.selectedUserGroupId) {
+      const options: CustomerSearchOptions = {
+        customerListId: this.selectedUserGroupId,
+        pageSize: pageSize,
+        currentPage: this.currentPage,
+      };
+      if (this.sortCode) {
```

Comment:
if `this.sortCode` is empty string `''`, it is also false. So, each time, we select a customer group, the sort is always `byNameAsc`

Reply:
The sorting depends on the customer group for example if the customer group id is 'A' then we only can sort by 'date' and if the customer group id is 'B' then we only can sort by 'name'.  If wrong sorting is passed then the backend ignores and send back us the default sorting.  so we removed the default sort (byNameAsc) if the customer group is passed in occ-asm.adapter.ts